### PR TITLE
Implement "Save Idea" Functionality #913

### PR DESCRIPTION
**Pull request by Rutuja177**
Implement "Save Idea" Functionality

**Description**:
This pull request introduces the "Save Idea" functionality to allow users to save ideas. The key changes and additions are as follows:

1. Created a "SavedIdea" collection to represent the relationship between users and saved ideas.
2. Added relation fields for "users" (many-to-many) and "Idea_card" (one-to-one) in the "SavedIdeas" collection.
3. Included a "SavedAt" timestamp to track when an idea was saved.

Thank you!

### CODE REVIEW

File: src/api/save-idea/content-types/save-idea/schema.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "draftAndPublish": true
+  },
+  "pluginOptions": {},
+  "attributes": {
+    "ObjectId": {
+      "type": "uid"
+    },
+    "users": {
+      "type": "relation",
+      "relation": "manyToMany",
```
Comment (dbradham) [inline src/api/save-idea/content-types/save-idea/schema.json:20]: Should this be oneToOne instead?

File: src/api/save-idea/content-types/save-idea/schema.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      "target": "plugin::users-permissions.user",
+      "inversedBy": "save_ideas"
+    },
+    "SavedAt": {
+      "type": "datetime"
+    },
+    "idea_card": {
```
Comment (dbradham) [inline src/api/save-idea/content-types/save-idea/schema.json:27]: if we already have the `ObjectId`, is this necessary?

Comment (Rutuja177) [reply src/api/save-idea/content-types/save-idea/schema.json]: The updates have been implemented.

File: src/api/save-idea/content-types/save-idea/schema.json (hunk 1 above)
Comment (Rutuja177) [reply src/api/save-idea/content-types/save-idea/schema.json]: The updates have been implemented.

**Review by dbradham — APPROVED** [review]
Looks good!

**Comment by Miminiverse** [comment]
Can we be more explicit in the collection's displayName instead of Save? Or the name is used for a certain purpose?

**Review by chungthuang — APPROVED** [review]

**Pull request merged**